pyirf/interpolation/tests/test_moment_morph_interpolator.py

```diff
@@ -0,0 +1,872 @@
+import numpy as np
+import pytest
+from scipy.stats import norm
+
+
+def expected_mean(a, b):
+    return 5 + (a / 5) + (b / 15)
+
+
+def expected_std(a, b):
+    return 1 + 0.05 * (a + b)
+
+
+@pytest.fixture
+def bins():
+    return np.linspace(-10, 40, 1001)
+
+
+@pytest.fixture
+def simple_1D_data(bins):
+    grid = np.array([[20], [40]])
+    target = np.array([30])
+    bin_contents = np.array(
+        [
+            np.diff(norm(loc=expected_mean(x, 0), scale=expected_std(x, 0)).cdf(bins))
+            for x in grid
+        ]
+    )
+
+    # Assert for probability outside binning
+    bin_contents /= bin_contents.sum(axis=1)[:, np.newaxis]
+
+    truth = np.diff(
+        norm(loc=expected_mean(target, 0), scale=expected_std(target, 0)).cdf(bins)
+    )
+    truth /= truth.sum()
+
+    return {
+        "grid": grid,
+        "target": target,
+        "bin_contents": bin_contents,
+        "truth": truth,
+    }
+
+
+@pytest.fixture
+def simple_2D_data(bins):
+    grid = np.array([[20, 20], [60, 20], [40, 60]])
+    target = np.array([25, 25])
+    bin_contents = np.array(
+        [
+            np.diff(
```

Comment:
There is a lot of repetition here, seems to be easy to factor out to something like

```
def _binned_pdf(dist, bins):
     return np.diff(dist.cdf(bins))
```
?

Reply:
Yeah, thats right. Doing so actually got rid of some 250 lines of code :sweat_smile: